wagtail/admin/widgets/chooser.py

```diff
@@ -171,9 +170,12 @@ def render(self, name, value, attrs=None, renderer=None):
     def base_js_init_options(self):
         """The set of options to pass to the JS initialiser that are constant every time this widget
         instance is rendered (i.e. do not vary based on id / name / value)"""
-        return {
+        opts = {
             "modalUrl": self.get_chooser_modal_url(),
         }
+        if self.linked_fields:
+            opts["linkedFields"] = self.linked_fields
```

Comment:
Tried to add tests for this in 91a26446c39dcd033fd3d174bb3cb9808b30726a and for trying it out with the testapp in 97bebfdb3db228ddd5f695d88f32dc9d54344aea. Might be nice to cherry-pick them (or at least the former, as I haven't written unit tests for the latter).

Reply:
Have cherry-picked the first one in bc7a711ab362a1ecdcf91632d6d530d8a4195f78 - not clear on what's being added by the second one that isn't already being tested in the first, but if you can give me some pointers to what you had in mind, I'd be happy to have a go at extending it.

Reply:
I guess it's just that the first one only tests the telepath adapter and the ones you wrote only test the views, so we don't really have a test that ensures it all works when you put the two of them together. That said, that's basically just how unit tests are and I don't think we can write a test for that without an automated browser.